src/visynApp/VisynApp.tsx

```diff
@@ -0,0 +1,56 @@
+import { AppShell, AppShellProps } from '@mantine/core';
+import * as React from 'react';
+import { JSXElementConstructor, ReactElement } from 'react';
+import { VisynHeader } from './header/VisynHeader';
+import { VisynLoginMenu } from './login/LoginMenu';
+import { useInitVisynApp } from './useInitVisynApp';
+
+/**
+ *
+ * @param header Optional custom header to be passed to the AppShell. If not provided, will use an empty VisynHeader.
+ * @param navbar Optional navbar component to be passed to AppShell.
+ * @param aside Optional aside component to be passed to AppShell
+ * @param footer Optional footer component to be passed to AppShell
+ * @param appShellProps Optional props to be passed directly to AppShell
+ * @param loginMenu Optional custom login menu. If not passed, will default to the VisynLoginMenu.
+ * @param headerHeight Optional height for the header, so that you can properly use 100% inside of your application. Does not set the height of the header, just calculates height elsewhere based on this number
+ * @param appName Name of application. Used in default login menu and header.
+ * @returns
+ */
+export function VisynApp({
+  header = null,
+  navbar = null,
+  aside = null,
+  footer = null,
+  appShellProps = null,
+  children,
+  appName,
+  headerHeight = 0,
+  loginMenu = <VisynLoginMenu watch appName={appName} />,
+}: {
+  header?: ReactElement<unknown, string | JSXElementConstructor<unknown>>;
+  navbar?: ReactElement<unknown, string | JSXElementConstructor<unknown>>;
+  aside?: ReactElement<unknown, string | JSXElementConstructor<unknown>>;
+  footer?: ReactElement<unknown, string | JSXElementConstructor<unknown>>;
+  appShellProps?: Partial<AppShellProps & React.RefAttributes<HTMLDivElement>>;
+  loginMenu?: JSX.Element;
+  children?: React.ReactChild;
+  headerHeight?: number;
+  appName: string;
+}) {
+  const { status } = useInitVisynApp();
+
+  return status === 'success' ? (
+    <AppShell
+      styles={{ root: { height: '100%' }, body: { height: `calc(100% - ${headerHeight}px)` }, main: { minHeight: '0px' } }}
```

Comment:
I would leave the definition of the styles to app, because you don't need this for example in [Sightline](https://github.com/datavisyn/sightline/blob/d7f67b1aebd92fe40d135209452fb04e9a926526/src/app/visynApp/VisynApp.tsx#L41).

```jsx
<VisynApp styles={{ root: { height: '100%' }, body: { height: `calc(100% - ${headerHeight}px)` }, main: { minHeight: '0px' } }}></VisynApp>
```

Otherwise: Why is the headerHeight configurable, when there is a constant in VisynHeader.tsx?

```ts
const HEADER_HEIGHT = 50;
```

Then the variable `headerHeight` can be removed and set to 50, right?

Reply:
The idea is by default we use the VisynHeader, but since users can override the header with any component they need, they should also be able to pass a custom header height. Now the next question, "if its a custom header, why do you need its height?". Great question! this is something weird about mantine (or maybe I just don't understand), but if you don't do this calculation for the height that I do on the body styles your components with normal 100% height will overflow. This is actually also why I included it in here, since I believe all applications will need something similar, and it might be hard to figure out why, as it took me a while in sightline.